modules/local/kmer_count_dim_reduction.nf

```diff
@@ -0,0 +1,62 @@
+process KMER_COUNT_DIM_REDUCTION {
+
+    tag "$meta.id"
+    label 'process_medium'
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-ac95cc1cb32439236d915b38af3e056ce8eb0375:2dc02878e5657bb1f787431c323c9d261fc6d520-0' :
+        'biocontainers/mulled-v2-ac95cc1cb32439236d915b38af3e056ce8eb0375:2dc02878e5657bb1f787431c323c9d261fc6d520-0' }"
+
+    input:
+    tuple val(meta), path(kmer_counts_file)
+    val dimensionality_reduction_method
+    val n_neighbors_setting
+    val autoencoder_epochs_count
+
+    output:
+    path '*_kmers_dim_reduction_embeddings.csv', emit: csv
+    path "versions.yml", emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def prefix = args.ext.prefix ?: "${meta.id}"
+    """
+
+    kmer_count_dim_reduction.py \\
+        $kmer_counts_file \\
+        ${prefix}_${dimensionality_reduction_method}_kmers_dim_reduction_embeddings.csv \\
+        --selected_methods $dimensionality_reduction_method \\
+        --n_neighbors_setting $n_neighbors_setting \\
+        --autoencoder_epochs_count $autoencoder_epochs_count
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        python: \$(python --version | sed 's/Python //g')
+        pandas: \$(python3 -c 'import pandas; print(pandas.__version__)')
+        tensorflow: \$(python3 -c 'import tensorflow; print(tensorflow.__version__)')
+        scikit-learn: \$(python3 -c 'import sklearn; sklearn.show_versions()')
```

Comment:
The `scikit-learn: \$(python3 -c 'import sklearn; sklearn.show_versions()')` part caused a crash when I tried running the subworkflow on my local computer.
Instead of outputting just a version number, it produced the long text below, which then caused a Nextflow crash with the message `"yaml.scanner.ScannerError: mapping values are not allowed here
  in "collated_versions.yml", line 22, column 11"`


```
System:
    python: 3.10.8 | packaged by conda-forge | (main, Nov 22 2022, 08:23:14) [GCC 10.4.0]
executable: /usr/local/bin/python3
   machine: Linux-5.4.0-166-generic-x86_64-with-glibc2.36

Python dependencies:
      sklearn: 1.1.2
          pip: 23.3.1
   setuptools: 68.2.2
        numpy: 1.24.4
        scipy: 1.11.3
       Cython: None
       pandas: 2.1.1
   matplotlib: 3.6.0
       joblib: 1.3.2
threadpoolctl: 3.2.0

Built with OpenMP: True

threadpoolctl info:
       user_api: blas
   internal_api: openblas
    num_threads: 8
         prefix: libopenblas
       filepath: /usr/local/lib/libopenblasp-r0.3.24.so
        version: 0.3.24
threading_layer: pthreads
   architecture: Haswell

       user_api: openmp
   internal_api: openmp
    num_threads: 8
         prefix: libgomp
       filepath: /usr/local/lib/libgomp.so.1.0.0
        version: None
```

Reply:
Before crashing at the software versions part, it counted the kmers and made CSV files for dimensionality reduction embeddings like it should, so at least that's good